### PR TITLE
Add NEAR account

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes https://github.com/trustwallet/wallet-core/issues/1115

### CODE REVIEW

**Review by vikmeup — APPROVED** [review]
Should derivation address be changed to a different format?

**Comment by hewigovens** [comment]
I thought the same but current one has checksum, not easy to make mistake

**Comment by vikmeup** [comment]
> I thought the same but current one has checksum, not easy to make mistake

but it will be compatible to receive on it from exchanges and wallets?

**Comment by hewigovens** [comment]
let me confirm with near team

**Review by madcake — APPROVED** [review]

File: src/NEAR/Account.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace TW::NEAR {
+
+class Account {
+public:
+    /// Determines whether a string makes a valid NEAR account id.
+    static bool isValid(const std::string& string);
+};
+
+} // namespace TW::NEAR
```
Comment (catenocrypt) [inline src/NEAR/Account.h:19]: Indent after namespace just like any other {} ?

File: src/NEAR/Account.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+using namespace TW;
+using namespace TW::NEAR;
+
+bool Account::isValid(const std::string& string) {
+    // https://docs.near.org/docs/concepts/account#account-id-rules
+    if (string.size() < 2 || string.size() > 64) {
+        return false;
+    }
+    std::regex pattern(R"(^(([a-z\d]+[\-_])*[a-z\d]+\.)*([a-z\d]+[\-_])*[a-z\d]+$)");
```
Comment (catenocrypt) [inline src/NEAR/Account.cpp:19]: Let's make pattern variable const.  Better even static.

**Review by catenocrypt — APPROVED** [review]

**Comment by catenocrypt** [comment]
Great work, minor comments added

**Comment by ilblackdragon** [comment]
Would be great to change this to account ids: https://nomicon.io/DataStructures/Account.html#account-id
Specifically, Trust Wallet can use https://nomicon.io/DataStructures/Account.html#implicit-account-creation to generate new account ids.

**Review by madcake — APPROVED** [review]

**Review by vikmeup — APPROVED** [review]

**Pull request merged**